atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,264 @@
+                outputs.append(instance)
+
+        return outputs
+
+    def get_buildroot(self, worker_metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        buildroots = []
+
+        # force ordering to work around python2 to python3 bug
+        platforms = []
+        for platform in worker_metadatas:
+            platforms.append(platform)
+        platforms.sort()
+        for platform in platforms:
+            for instance in worker_metadatas[platform]['buildroots']:
+                instance['id'] = platform + '-' + repr(instance['id'])
+                buildroots.append(instance)
+
+        return buildroots
+
+    def get_build(self, metadata, worker_metadatas):
+        start_time = int(atomic_reactor_start_time)
+
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
+
+        component = get_preferred_label(labels, 'com.redhat.component')
+        version = get_preferred_label(labels, 'version')
+        release = get_preferred_label(labels, 'release')
+
+        source = self.workflow.source
+        if not isinstance(source, GitSource):
+            raise RuntimeError('git source required')
+
+        extra = {'image': {'autorebuild': is_rebuild(self.workflow)}}
+        koji_task_id = metadata.get('labels', {}).get('koji-task-id')
+        if koji_task_id is not None:
+            self.log.info("build configuration created by Koji Task ID %s",
+                          koji_task_id)
+            try:
+                extra['container_koji_task_id'] = int(koji_task_id)
+            except ValueError:
+                self.log.error("invalid task ID %r", koji_task_id, exc_info=1)
+
+        fs_result = self.workflow.prebuild_results.get(AddFilesystemPlugin.key)
+        if fs_result is not None:
+            try:
+                fs_task_id = fs_result['filesystem-koji-task-id']
+            except KeyError:
+                self.log.error("%s: expected filesystem-koji-task-id in result",
+                               AddFilesystemPlugin.key)
+            else:
+                try:
+                    task_id = int(fs_task_id)
+                except ValueError:
+                    self.log.error("invalid task ID %r", fs_task_id, exc_info=1)
+                else:
+                    extra['filesystem_koji_task_id'] = task_id
+
+        # get the annotations from the worker BuildResponses
+        valid_help_file = False
+        help_file = None
+        for platform in worker_metadatas:
+            worker_annotations = get_worker_build_info(self.workflow, platform)
+            if worker_annotations and 'help_file' in worker_annotations:
+                valid_help_file = True
+                tmp_help_file = json.loads(worker_annotations['help_file'])
+                if tmp_help_file is not None:
```

Comment:
Oh, I see. Do we need the `if tmp_help_file is not None:` then? We could just always assign that variable to whatever `json.loads()` returns.

I wonder if it would be more readable like this:

```
help_metadata = {}
for ... in ...:
    if worker_annotations and 'help_file' in worker_annotations:
        help_metadata['help'] = json.loads(....)

extra['image'].update(help_metadata)
```